### PR TITLE
feat(application_modes): 11737 - add map mode to side bar

### PR DESCRIPTION
**Pull request by IlyaIzr**
In this reqquest i'm only creating modes entity and adding modes buttons to sidebar.
Expected task result: 
image
Actual result:
image

This PR wil not be merged until other task will be completed. So in next PR's i will be working at adopting `url_encoder`, creating new toolbox panel and moving some panels around according to our design

P.s. i will merge main in here soon

### CODE REVIEW

File: src/features/toolbar/components/Toolbar.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  return (
+    <ActionsBar>
+      {sortByPredefinedOrder(Object.values(controls), controlsOrder).map(
+        (control) => (
+          <div key={nanoid(4)} className={s.sideBarContainer}>
+            <div
+              className={s.buttonWrap}
+              onClick={() =>
+                control.onClick && control.onClick(!control.active)
```
Comment (propakov) [inline src/features/toolbar/components/Toolbar.tsx]:
```suggestion
              onClick={() => control.onClick?.(!control.active)}
```

File: src/features/side_bar/components/SideBar/SideBar.module.css [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
   z-index: 1;
   display: inline-block;
-  width: 48px;
-  text-align: center;
-  margin: var(--unit) auto;
+  /* width: 48px; */
+  /* text-align: center; */
+  padding: 0 var(--unit);
+  margin: var(--unit) 0;
+  text-decoration: none;
```
Comment (propakov) [inline src/features/side_bar/components/SideBar/SideBar.module.css:7]: text-decoration can be removed as it's used from '<a>' tag

**Review by propakov — APPROVED** [review]
Good job! 👍

File: src/features/side_bar/components/SideBar/SideBar.module.css (hunk 2 above)
Comment (IlyaIzr) [reply src/features/side_bar/components/SideBar/SideBar.module.css:7]: it's needed! else the text becomes underlined because of routers <Link> component

Comment (propakov) [reply src/features/side_bar/components/SideBar/SideBar.module.css:7]: Screenshot 2022-08-26 at 09 03 12
Screenshot 2022-08-26 at 09 03 34

File: src/features/toolbar/components/Toolbar.tsx (hunk 1 above)
Comment (IlyaIzr) [reply src/features/toolbar/components/Toolbar.tsx]: that's how our formatter works 😆

File: src/features/side_bar/components/SideBar/SideBar.module.css (hunk 2 above)
Comment (IlyaIzr) [reply src/features/side_bar/components/SideBar/SideBar.module.css:7]: I still have underline without that style
image
This is what brings it
image
What tag are you talking about?

Comment (propakov) [reply src/features/side_bar/components/SideBar/SideBar.module.css:7]: Screenshot 2022-08-26 at 11 26 37

Comment (propakov) [reply src/features/side_bar/components/SideBar/SideBar.module.css:7]: so as you see style from 'a' tag is applied
anyway it's not that major, lets stay as it is, if you see underline

Comment (propakov) [reply src/features/side_bar/components/SideBar/SideBar.module.css:7]: @IlyaIzr I've found that one extension in my browser affect this style, sorry for this.

**Comment by IlyaIzr** [comment]
At this point I count this PR as reviewed and before merging it to main I will merge nesceserry add-ons to this branch

**Pull request merged**